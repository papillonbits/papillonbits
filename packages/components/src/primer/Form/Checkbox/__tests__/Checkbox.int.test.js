import { isChecked, isNotChecked, isCheckedInactive, isNotCheckedInactive } from './Checkbox.int.story'

describe('<Checkbox />', () => {
  afterEach(() => jest.clearAllMocks())

  describe('Render', () => {
    test('must match isChecked()', () => {
      expect(global.renderToJSON(isChecked())).toMatchSnapshot()
    })

    test('must match isCheckedInactive()', () => {
      expect(global.renderToJSON(isCheckedInactive())).toMatchSnapshot()
    })

    test('must match isNotChecked()', () => {
      expect(global.renderToJSON(isNotChecked())).toMatchSnapshot()
    })

    test('must match isNotCheckedInactive()', () => {
      expect(global.renderToJSON(isNotCheckedInactive())).toMatchSnapshot()
    })
  })
})
