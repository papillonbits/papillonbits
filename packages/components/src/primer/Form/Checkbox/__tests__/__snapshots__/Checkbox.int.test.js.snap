// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Render must match isChecked() 1`] = `
<div>
  <div
    class="m-0 form-checkbox"
  >
    <label
      for="checkbox"
    >
      <input
        aria-describedby="help-text-for-checkbox"
        checked=""
        type="checkbox"
      />
      is checked
    </label>
  </div>
</div>
`;

exports[`<Checkbox /> Render must match isCheckedInactive() 1`] = `
<div>
  <div
    class="m-0 form-checkbox checkbox-inactive"
  >
    <label
      for="checkbox"
    >
      <input
        aria-describedby="help-text-for-checkbox"
        checked=""
        type="checkbox"
      />
      is checked
    </label>
  </div>
</div>
`;

exports[`<Checkbox /> Render must match isNotChecked() 1`] = `
<div>
  <div
    class="m-0 form-checkbox"
  >
    <label
      for="checkbox"
    >
      <input
        aria-describedby="help-text-for-checkbox"
        type="checkbox"
      />
      is not checked
    </label>
  </div>
</div>
`;

exports[`<Checkbox /> Render must match isNotCheckedInactive() 1`] = `
<div>
  <div
    class="m-0 form-checkbox checkbox-inactive"
  >
    <label
      for="checkbox"
    >
      <input
        aria-describedby="help-text-for-checkbox"
        type="checkbox"
      />
      is not checked
    </label>
  </div>
</div>
`;
